Impossible-Agent-Suite/staxmigrate/src/agent.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  Finding,
+  FindingType,
+  FindingSeverity,
+  TransactionEvent,
+  HandleTransaction
+} from 'forta-agent';
+
+const IF_ADDR = '0xb0e1fc65c1a741b4662b813eb787d369b8614af1';
+const ZERO_ADDR = '0x0000000000000000000000000000000000000000';
+
+export const IF_ABI = [
+  'event Transfer(address indexed from, address indexed to, uint256 value)',
+  'function staxMigrate(uint amount)'
+];
+
+export const provideHandleTransaction = (address: string): HandleTransaction => {
+  return async (tx: TransactionEvent): Promise<Finding[]> => {
+    // Setup the findings array
+    const findings: Finding[] = [];
+
+    // Get all times `staxMigrate` was called
+    const migrations = tx.filterFunction(IF_ABI[1], address);
+    // If there were calls to the `staxMigrate` function 
+    if(migrations.length) {
+      // Get all times `Transfer` was emitted
+      const transfers = tx.filterLog(IF_ABI[0], address);
+      // For each `staxMigrate` call
+      migrations.forEach((migration) => {
+        // For each `Transfer` event
+        transfers.forEach((transfer) => {
+          // Check if the `Transfer` event is a mint
+          if(transfer.args.from == ZERO_ADDR) {
+            // Get the amount of STAX tokens that are going 'in'
+            const staxAmountIn = migration.args.amount.toString();
+            const ifAmountOut = transfer.args.value.toString();
+            // If `staxAmountIn` and `ifAmountOut` are different 
+            if(staxAmountIn != ifAmountOut) {
```

Comment:
This will not work for txns with multiple migrations. Lets say 2 valid migrations occur with different amounts, this logic will generate two findings.
You should do something like a two pointers algorithm to find a match for each `staxMigrate` call